caPutLogApp/caPutLogTask.c

```diff
@@ -145,46 +146,75 @@ int caPutLogTaskStart(int config)
         }
     }
 
+    caPutLogConfig = config;
+
     if (epicsThreadGetId("caPutLog")) {
+#if 0
```

Comment:
I thought the same about a debug flag but did not want to do too many changes that have nothing to do with my enhancements. In my code I normally have a `xxxDebug` variable that can be set from the ioc shell to allow switching on debugging at run time instead of compile time.

Reply:
Personally, I have deprecated compile-time debug switches and I passive aggressively remove them when I have the opportunity.
* They often break the build if they are not tested enough (CI can help, obviously).
* They have a tendency to not be in the code when you need them.
* They often don't play well with software QA / static code analysis.

Reply:
`caPutLogDebug` run time variable added.